crates/red_knot_python_semantic/src/types.rs

```diff
@@ -362,6 +364,10 @@ impl<'db> Type<'db> {
                 // TODO defer to Type::Instance(<bytes from typeshed>).member
                 Type::Unknown
             }
+            Type::TupleLiteral(_) => {
+                // TODO: defer to `builtins.tuple` methods from typeshed's stubs
```

Comment:
We won't actually want to do this for all or even most methods. For example, we will want to special-case `__getitem__` to return a special callable type which knows exactly which type to return (or raises an error) for each index it might be called with. We'll similarly need to special case many other methods: `__len__`, `__contains__`, `__iter__`, `count`, `index`, possibly some others as well.
```suggestion
                // TODO: implement tuple methods
```